JetNews/app/src/main/java/com/example/jetnews/ui/article/ArticleScreen.kt

```diff
@@ -131,20 +137,45 @@ fun ArticleScreen(
         topBar = {
             InsetAwareTopAppBar(
                 title = {
-                    Text(
-                        text = "Published in: ${post.publication?.name}",
-                        style = MaterialTheme.typography.subtitle2,
-                        color = LocalContentColor.current
-                    )
+                    Row(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .wrapContentWidth(align = Alignment.CenterHorizontally)
+                            .padding(start = 30.dp)
+                    ) {
+                        Surface(
+                            color = Color(0xff073042),
+                            shape = CircleShape,
+                            modifier = Modifier.size(36.dp, 36.dp)
+                        ) {
+                            Image(
+                                painter = painterResource(R.drawable.ic_article_icon),
+                                contentDescription = "",
+                                colorFilter = ColorFilter.tint(Color(0xff32dd84)),
+                                modifier = Modifier.padding(4.dp)
+                            )
+                        }
+                        Text(
+                            text = stringResource(id = R.string.published_in, post.publication?.name ?: ""),
+                            style = MaterialTheme.typography.subtitle2,
+                            color = LocalContentColor.current,
+                            modifier = Modifier
+                                .padding(start = 10.dp)
+                                .weight(1.5f)
+                        )
+                    }
                 },
                 navigationIcon = {
                     IconButton(onClick = onBack) {
-                        Icon(
+                        Image(
                             imageVector = Icons.Filled.ArrowBack,
-                            contentDescription = stringResource(R.string.cd_navigate_up)
+                            contentDescription = stringResource(R.string.cd_navigate_up),
+                            colorFilter = ColorFilter.tint(MaterialTheme.colors.primary)
                         )
                     }
-                }
+                },
+                elevation = 0.dp,
+                backgroundColor = MaterialTheme.colors.onPrimary
```

Comment:
`onPrimary` is a color meant for Text that is shown on top of a Primary surface. I don't think that's what you mean here?

Reply:
I replaced it with  MaterialTheme.colors.surface

Reply:
Added elevation on scroll